coreboot-utils: nvramtool fails to build on aarch64-linux

### Nixpkgs version

- Unstable (26.05)

### Describe the bug

nvramtool has failed to build on aarch64-linux since 2024-12-02 https://hydra.nixos.org/build/280731687 was the last good build, [current builds](https://hydra.nixos.org/job/nixos/unstable/nixpkgs.nvramtool.aarch64-linux) are failing with this error:
```
gcc -O2 -g -Wall -Wextra -Wmissing-prototypes -I. -DCMOS_HAL=1   -c -o accessors/cmos-hw-unix.o accessors/cmos-hw-unix.c
accessors/cmos-hw-unix.c: In function 'cmos_hal_read':
accessors/cmos-hw-unix.c:179:9: error: 'return' with no value, in function returning non-void [-Wreturn-mismatch]
  179 |         return;
      |         ^~~~~~
accessors/cmos-hw-unix.c:177:22: note: declared here
  177 | static unsigned char cmos_hal_read(__attribute__((unused)) unsigned index)
      |                      ^~~~~~~~~~~~~
make: *** [<builtin>: accessors/cmos-hw-unix.o] Error 1
```

### Steps to reproduce

- [check hydra build history for nixpkgs.nvramtool.aarch64-linux](https://hydra.nixos.org/job/nixos/unstable/nixpkgs.nvramtool.aarch64-linux)
- observe that it is all build failures

### Expected behaviour

the package should successfully build, or it should not include aarch64-linux in its supported platforms.

### Screenshots

_No response_

### Relevant log output

```console

```

### Additional context

_No response_

### System metadata

I don't have an aarch64-linux machine to test on at the moment.

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/misc/coreboot-utils/default.nix
+++ b/pkgs/tools/misc/coreboot-utils/default.nix
@@ -110,6 +110,10 @@ let
       pname = "nvramtool";
       meta.description = "Read and write coreboot parameters and display information from the coreboot table in CMOS/NVRAM";
       meta.mainProgram = "nvramtool";
+      meta.platforms = [
+        "x86_64-linux"
+        "i686-linux"
+      ];
     };
     superiotool = generic {
       pname = "superiotool";

